### PR TITLE
Added Gateway classes + Interface, updated design document, updated employees file

### PR DESCRIPTION
**Pull request by rosez19**
- Updated Design Document accessibility report (with Lucas' edits that somehow got deleted) and other sections
- Created ReadWriter gateway classes for EventManager and EmployeeManager to read/update files - since there were dependency issues (affecting clean architecture) I made interfaces that both of them implement
- Updated EventManager and EmployeeManager classes accordingly
- Added dates to employees.txt

**TODO:**
- Create UserManagerReadWriter class
- I think we should change the update method in EmployeeManager/EmployeeManagerReadWriter... I think it currently rewrites the entire employee.txt file when one employee has a change in unavailability and I think it should just update for that one employee. Also its current implementation doesn't work as it only writes in the unavailable dates for employees assigned to the new event

### CODE REVIEW

**Comment by h4kureii** [comment]
It would be far more inefficient to rewrite individual Employees when rewriting large number of Employees (which our program frequently does), so I think the implementation should stay as is.

**Comment by rosez19** [comment]
> It would be far more inefficient to rewrite individual Employees when rewriting large number of Employees (which our program frequently does), so I think the implementation should stay as is.

Yes ok you're right! I didn't read the code properly lol

**Review by h4kureii — CHANGES_REQUESTED** [review]
Small things:
I think we should move employees_path into the EmployeeManagerReadWriter; pretty sure the gateway class is meant to separate read/writing from the use case so it wouldn't make sense to include the file path in the use case.
EventNotFoundError is outdated; see main branch (EventManager should not hold one (not sure why was there to begin with), and it's renamed/refactored to EventNotFoundException to keep in line with naming conventions)

otherwise ok (i think)

prefer this implementation of EventManager gateway over Lucas'

**Comment by rosez19** [comment]
> Small things: I think we should move employees_path into the EmployeeManagerReadWriter; pretty sure the gateway class is meant to separate read/writing from the use case so it wouldn't make sense to include the file path in the use case. EventNotFoundError is outdated; see main branch (EventManager should not hold one (not sure why was there to begin with), and it's renamed/refactored to EventNotFoundException to keep in line with naming conventions)
> 
> otherwise ok (i think)
> 
> prefer this implementation of EventManager gateway over Lucas'

Ok fixed!

File: phase2/Design Document.md [hunk 1]
```diff
@@ -133,7 +158,6 @@ employees and the saved User data, so those would be packaged into one folder as
 
 ---
 ## Design Patterns
-### Command Design pattern
```
Comment (ZS106) [inline phase2/Design Document.md:136]: maybe keep this title?

**Review by h4kureii — APPROVED** [review]
We'll probably have to make changes to EventManager/EventManagerReadWriter when we implement the gateway for UserManager but this is good

**Review by ZS106 — APPROVED** [review]

**Pull request merged**